client.go

```diff
@@ -176,7 +207,15 @@ func (c *Client) BulkIdentify(ctx context.Context, batch []*IdentityTraits) erro
 // GetEnvironmentFlagsFromAPI tries to contact the Flagsmith API to get the latest environment data.
 // Will return an error in case of failure or unexpected response.
 func (c *Client) GetEnvironmentFlagsFromAPI(ctx context.Context) (Flags, error) {
-	resp, err := c.client.NewRequest().
+	req := c.client.NewRequest()
+	maybeEc := ctx.Value(contextKeyEvaluationContext)
+	if maybeEc != nil {
+		envCtx := maybeEc.(*EvaluationContext).Environment
+		if envCtx != nil {
+			req.SetHeader(flaghttp.EnvironmentKeyHeader, envCtx.APIKey)
```

Comment:
Why are we setting this header again?

Reply:
We can provide an `EvaluationContext` to `GetFlags` that has a different API key.